test/blackbox-tests/test-cases/melange/emit-with-runtime-deps-dir-target.t

```diff
@@ -0,0 +1,45 @@
+Test simple interactions between melange.emit and copy_files
+
+  $ cat > dune-project <<EOF
+  > (lang dune 3.7)
+  > (using melange 0.1)
+  > (using directory-targets 0.1)
+  > EOF
+
+  $ cat > dune <<EOF
+  > (rule (target (dir some_dir))
+  >  (action
+  >   (progn (bash "mkdir %{target}")
+  >    (bash "echo hello from file inside dir target > %{target}/inside-dir-target.txt"))))
+  > (melange.emit
+  >  (alias mel)
+  >  (target output)
+  >  (runtime_deps ./some_dir)
+  >  (module_system commonjs))
+  > EOF
+
+  $ cat > main.ml <<EOF
+  > let dirname = [%bs.raw "__dirname"]
+  > let file_path = "./some_dir/inside-dir-target.txt"
+  > let file_content = Node.Fs.readFileSync (dirname ^ "/" ^ file_path) \`utf8
+  > let () = Js.log file_content
+  > EOF
+
+Rules created for the assets in the output directory
+
+  $ mkdir output
+  $ dune rules @mel | grep file.txt
+  [1]
+
+  $ dune build @mel --display=short
+          melc .output.mobjs/melange/melange__Main.{cmi,cmj,cmt}
+          melc output/main.js
+  File "dune", line 5, characters 0-97:
+  5 | (melange.emit
+  6 |  (alias mel)
+  7 |  (target output)
+  8 |  (runtime_deps ./some_dir)
+  9 |  (module_system commonjs))
+  Error: Error trying to read targets after a rule was run:
+  - output/some_dir: Unexpected file kind "S_DIR" (directory)
+  [1]
```

Comment:
With copy instead of symlink, the error is:

```
+  Error: Is a directory
+  -> required by _build/default/output/some_dir
+  -> required by alias output/mel
```

Reply:
Interesting. So this is only a problem with symlinking? I would expect the same problem with copying. For symlinking, we have `Action_builder.symlink_dir` to get around this.

Reply:
It looks like a problem with copying too, right? where copying doesn't pass `-r`

Reply:
my code is probably missing a directory check, and call `with_targets ~directory_targets...`  rather than the current form. 

Which brings up the question: what's the best way to detect whether I'm looking at a directory or a file in an action builder monad?

Reply:
> For symlinking, we have `Action_builder.symlink_dir` to get around this.

What would happen if user tries to copy a dir and also another dir inside it?

Reply:
I believe directory targets are opaque, so that works.

Reply:
Not sure if there's a good way to check . For now, I'd do the following:

* Use symlinking
* Document that directories are a known limitation